python/src/iceberg/expressions/base.py

```diff
@@ -285,6 +270,9 @@ def __str__(self):
     def __repr__(self):
         return f"BoundReference(field={repr(self.field)}, accessor={repr(self._accessor)})"
 
+    def __eq__(self, other) -> bool:
```

Comment:
If you convert this to a `@dataclass`, I think you'd get these for free.

Reply:
changed to dataclasses